tests/test_version_range.py

```diff
@@ -209,6 +211,32 @@ def test_PypiVersionRange_raises_ivr_for_invalid_ranges(self):
         except InvalidVersionRange as ivre:
             assert str(ivre).startswith("Unsupported character")
 
+    def test_NpmVersionRange_from_native(self):
+        """
+        Tests for NpmVersionRange.from_native
+        """
+        # test for `^version` or `Compatible with version`
+        npm_range = "^1.2.9"
+        npm_range_str = "vers:npm/>=1.2.9|<2.0.0"
+        expected = (
+            VersionConstraint(comparator=">=", version=SemverVersion(string="1.2.9")),
+            VersionConstraint(comparator="<", version=SemverVersion(string="2.0.0")),
+        )
+        version_range = NpmVersionRange.from_native(npm_range)
+        assert version_range.constraints == expected
+        assert version_range.to_string() == npm_range_str
+
+        # test for `~version` or `Approximately equivalent to version`
+        npm_range = "~3.8.2"
+        npm_range_str = "vers:npm/>=3.8.2|<3.9.0"
+        expected = (
+            VersionConstraint(comparator=">=", version=SemverVersion(string="3.8.2")),
+            VersionConstraint(comparator="<", version=SemverVersion(string="3.9.0")),
+        )
+        version_range = NpmVersionRange.from_native(npm_range)
+        assert version_range.constraints == expected
+        assert version_range.to_string() == npm_range_str
+
 
 VERSION_RANGE_TESTS_BY_SCHEME = {
     "nginx": ["0.8.40+", "0.7.52-0.8.39", "0.9.10", "1.5.0+, 1.4.1+"],
```

Comment:
As you're already working on npm version range, how about adding an entry here too :)